modules/templates/htmlrenderer.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package templates
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/watcher"
+
+	"github.com/unrolled/render"
+)
+
+var rendererKey interface{} = "templatesHtmlRendereer"
+
+// HTMLRenderer returns the current html renderer for the context or creates and stores one within the context for future use
+func HTMLRenderer(ctx context.Context) (context.Context, *render.Render) {
+	rendererInterface := ctx.Value(rendererKey)
+	if rendererInterface != nil {
+		renderer, ok := rendererInterface.(*render.Render)
+		if ok && renderer != nil {
```

Comment:
```suggestion
		if ok {
```
Or why are you nil-checking twice?